webapp/lib/controller.dart

```diff
@@ -3,6 +3,8 @@ library controller;
 import 'dart:async';
 import 'dart:collection';
 
+import 'package:firebase/firebase.dart';
```

Comment:
Nit: 
```suggestion
import 'package:firebase/firebase.dart' show FirebaseError;
```
(let's narrow down the import, since we shouldn't really use the firebase package in this file, but we now need to import it for the firebase error - this is similar to what we do for the `PubSubException`)

Reply:
Nice suggestion. TIL